src/main/webapp/dev/js/main/feedbackSubmissionsEdit.js

```diff
@@ -1156,16 +1222,50 @@ function prepareRankQuestions() {
 
         const isRankingRecipients = $(`#rankToRecipients-${qnNum}`).val() === 'true';
 
+        // Check if feedback session is open or under preview mode
         if (!$('#response_submit_button').is(':disabled')
             || isPreview()) {
+            const areDuplicateRanksAllowed = $(`#rankAreDuplicatesAllowed-${qnNum}`).val() === 'true';
```

Comment:
umm i can't see html being generated here. If you mean line 1238 and 1239 i had this discussion with @wenmogu while working on [8836](https://github.com/TEAMMATES/teammates/pull/8836). It was suggested that all dynamic html generation should be preferably done using js since JSPs represent general structure of a web page and adding such logic to JSP would make it very specific to a particular question type.

Issue comment [here](https://github.com/TEAMMATES/teammates/pull/8836/commits/c5263a249a75471f14aa5cba54f9a99119117d86#r190470394)

![image](https://user-images.githubusercontent.com/11634699/44003683-5e61d8ba-9e74-11e8-8d10-989e84a92955.png)